pkg/kritis/cryptolib/public_key.go

```diff
@@ -55,12 +55,18 @@ func NewPublicKey(keyType KeyType, keyData []byte, keyID string) (*PublicKey, er
 			return nil, err
 		}
 		newKeyID = id
+		if signatureAlgorithm != UndefinedSigningAlgorithm {
```

Comment:
For code readability reasons, I'm hesitant to say that all pgp keys have undefined signing algorithm going forward. 

If we need this for backwards compatibility, maybe we can support both undefined and a new "pgp" signature algorithm. Someone who is more knowledgeable about kirtis should weigh in.

Reply:
I think it would make sense to let PGP keys have a new PGP SignatureAlgorithm. 

Perhaps it makes sense to split `NewPublicKey` by KeyType? You could do the following:
```
NewPgpPublicKey(keyData []byte, keyID string) (*PublicKey, error)
NewPkixPublicKey(signatureAlgorithm SignatureAlgorithm, keyData []byte, keyID string) (*PublicKey, error)
NewJwtPublicKey(signatureAlgorithm SignatureAlgorithm, keyData []byte, keyID string) (*PublicKey, error)
```

Reply:
If we add the PGP Signature algorithm there would be no need to split by KeyType right? Also, the idea behind using it as undefined was to highlight the fact that it was not used in this case. Would something like UnusedPGP work to ensure that it is understood that this is simply a placeholder?

Another question would be are there future plans that would make this field necessary in the PGP case?

Reply:
As we discussed in standup a while back, for now just change this to UnusedPGP or PGPSigningAlgorithmNotApplicable

Reply:
Ditto to @alexcope . And I also think makes sense to split the constructors by key type, but it doesn't have to be done in this PR.

Reply:
done.